src/membership/members.rs

```diff
@@ -57,11 +60,36 @@ const DEFAULT_MAX_HANDLE_LENGTH: u32 = 40;
 const DEFAULT_MAX_AVATAR_URI_LENGTH: u32 = 1024;
 const DEFAULT_MAX_ABOUT_TEXT_LENGTH: u32 = 2048;
 
+pub type RoleId = u32;
```

Comment:
Ideally we want the type `BTreeMap<enum Role, ActorId>` to represent the members roles.
Unfortunately there is no implementation of the codec Encode/Decode traits for BTreeMap when the key is a n `enum`. To avoid implementing it and having to subsequently implement corresponding encoders and decoders in polkadot-js api/types libraries I went with a simpler approach of using a `u32` as the key, but keeping the `enum Role` to have "named roles".

I did consider dropping BTreeMap and just go with a `Vec<ActorInRole>`.

Regarding the named roles, its possible to get rid of the `enum Role`. There are several approaches, that involves passing in a type on the trait that handles mapping.

Reply:
Now I can't recall what the premise is correct here? `BTreeMap<Role, ActorId>` would imply that a member can only ever occupy a single role of a given type at any time, which is a constraint we do not wish to capture?